### PR TITLE
Prepare metadata for Hackage release

### PR DESCRIPTION
**Pull request by qaristote**
I consider the current version of the master branch to be good enough to be published on Hackage and integrated inside Pirouette. This PR thus makes the Cabal project Hackage-compliant.

- Is anything else needed before publishing this as version `0.1` on Hackage ?
- Is it a good idea to run `cabal check` as a CI action to ensure the project is always Hackage-compliant ?
- I only used the `SMT` category as the library has a very specific goal. Should other tags be added ?

### CODE REVIEW

File: smtlib-backends.cabal [hunk 1]
```diff
@@ -1,21 +1,30 @@
 name:                smtlib-backends
 version:             0.1
-synopsis:            A library providing low-level functions for SMT-LIB-based interaction with SMT solvers. 
-description:         This library provides an extensible interface for interacting with SMT solvers using SMT-LIB. It has built-in support for running solvers as external processes, and the smtlib-backends-z3 package provides a backend that uses inlined call to Z3's C API.
+synopsis:            Low-level functions for SMT-LIB-based interaction with SMT solvers. 
+description:         This library provides an extensible interface for interacting with SMT solvers using SMT-LIB. It has built-in support for running solvers as external processes, and the smtlib-backends-z3 package provides a backend that uses inlined calls to Z3's C API.
```
Comment (facundominguez) [inline smtlib-backends.cabal:4]: This line is too long to read.

**Review by facundominguez — APPROVED** [review]
LGTM

You might want to add a ChangeLog

The readme can use some tweaks but I'll open other issues for that.

File: smtlib-backends.cabal (hunk 1 above)
Comment (qaristote) [reply smtlib-backends.cabal]: The changelog field will only be useful starting with version `0.2` though, will it not ?

Comment (facundominguez) [reply smtlib-backends.cabal]: Yeah. It would be there only to provide the scaffolding for when it is needed.

**Pull request merged**